src/command/backup/backup.c

```diff
@@ -489,147 +489,151 @@ backupBuildIncr(const InfoBackup *infoBackup, Manifest *manifest, Manifest *mani
-    if (resumeData->manifestParentName == NULL && strEqZ(info->name, BACKUP_MANIFEST_FILE INFO_COPY_EXT))
-        FUNCTION_TEST_RETURN_VOID();
+            // Skip backup.manifest.copy -- it must be preserved to allow resume again if this process throws an error before
+            // writing the manifest for the first time
+            if (manifestParentName == NULL && strEqZ(info.name, BACKUP_MANIFEST_FILE INFO_COPY_EXT))
+                continue;
 
-    // Build the name used to lookup files in the manifest
-    const String *manifestName = resumeData->manifestParentName != NULL ?
-        strNewFmt("%s/%s", strZ(resumeData->manifestParentName), strZ(info->name)) : info->name;
+            // Build the name used to lookup files in the manifest
+            const String *manifestName = manifestParentName != NULL ?
+                strNewFmt("%s/%s", strZ(manifestParentName), strZ(info.name)) : info.name;
 
-    // Build the backup path used to remove files/links/paths that are invalid
-    const String *backupPath = strNewFmt("%s/%s", strZ(resumeData->backupPath), strZ(info->name));
+            // Build the backup path used to remove files/links/paths that are invalid
+            const String *const backupPath = strNewFmt("%s/%s", strZ(backupParentPath), strZ(info.name));
 
-    // Process file types
-    switch (info->type)
-    {
-        // Check paths
-        // -------------------------------------------------------------------------------------------------------------------------
-        case storageTypePath:
-        {
-            // If the path was not found in the new manifest then remove it
-            if (manifestPathFindDefault(resumeData->manifest, manifestName, NULL) == NULL)
-            {
-                LOG_DETAIL_FMT("remove path '%s' from resumed backup", strZ(storagePathP(storageRepo(), backupPath)));
-                storagePathRemoveP(storageRepoWrite(), backupPath, .recurse = true);
-            }
-            // Else recurse into the path
+            // Process file types
+            switch (info.type)
             {
-                BackupResumeData resumeDataSub = *resumeData;
-                resumeDataSub.manifestParentName = manifestName;
-                resumeDataSub.backupPath = backupPath;
+                // Check paths
+                // -----------------------------------------------------------------------------------------------------------------
+                case storageTypePath:
+                {
+                    // If the path was not found in the new manifest then remove it
+                    if (manifestPathFindDefault(manifest, manifestName, NULL) == NULL)
+                    {
+                        LOG_DETAIL_FMT("remove path '%s' from resumed backup", strZ(storagePathP(storageRepo(), backupPath)));
+                        storagePathRemoveP(storageRepoWrite(), backupPath, .recurse = true);
+                    }
+                    // Else recurse into the path
```

Comment:
The "else" is missing from the code.

Reply:
Yep, that's in main as well. Not an issue since there will not be a path to recurse into, but not as efficient as it should be.

Reply:
Fixed in b54d25cf7a.